### DISCUSSION TITLE
Update TileMap to optionally use dual-grid

### DISCUSSION THREAD

**Discussion by Jearil** [godotengine/godot-proposals#10422] [2D]
Oskar Stålberg [proposed a system](https://x.com/OskSta/status/1448248658865049605) to use a duel-grid in order to reduce the number of tiles needed for auto-tiling in a tilemap. [@Jesscodes](https://www.youtube.com/@jesscodes) created a [video demonstration](https://www.youtube.com/watch?v=jEWFSv3ivTg) of this technique and included a [sample project](https://github.com/jess-hammer/dual-grid-tilemap-system-godot) of it working in Godot.

With the technique, you can make a rounded corner, low tile, auto-tiling grid system. It uses only 16 tiles rather than 47.

While one can use her code to implement it in their own projects, top level implementation into Godot itself seems like it would be extremely useful.

**Comment by ivandardi** [comment]
Yes, I agree with this.

The video demonstration didn't mention if there were any downsides to using the dual-grid system. It would be nice if someone more experienced could bring up a list of pros and cons of the dual-grid system just to make sure that this is a good thing to implement.

**Comment by JohnBakhmat** [comment]
I thought we might be able to make it without code with current terrain setup but nah. Awesome idea tho
